docs/source/reference/api-server/examples/api-server-auth-proxy.rst

```diff
@@ -0,0 +1,127 @@
+    :width: 70%
+
+Copy and paste the token into the terminal to save the auth for the SkyPilot CLI.
+
+.. code-block:: console
+
+    ...
+
+    http://a.b.c.d/token
+
+    Paste the token: eyJfb2F1dGgyX3Byb3h5IjogInVYcFRTTGZGSEVYeHVGWXB2NEc4dHNKTzVET253YkRVVEJ5SkVFM1cxYkg1V29TQVhSRk4tLXg1NFotT1hab0ZsV1BMUEJicTE2NXZmZmdWQ0FrVnQtMktlM0hpenczOWhLLTRMZ3...
+    Logged into SkyPilot API server at: http://a.b.c.d
+    └── Dashboard: http://a.b.c.d/dashboard
+
+
+This will copy the relevant auth cookies from the browser into the CLI.
+
+.. _oauth2-proxy-okta:
+
+Setting up OAuth2 Proxy with Okta
+---------------------------------
+
+The SkyPilot API server helm chart can also deploy and configure `OAuth2 Proxy <https://oauth2-proxy.github.io/>`__ to provide an out-of-the-box auth proxy setup.
+
+Here's how to set it up:
+
+Create application in Okta
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+From your Okta admin panel, navigate to **Applications > Applications**, then click the **Create App Integration** button.
+
+* For **Sign-in method**, choose **OIDC - OpenID Connect**
+* For **Application type**, chose **Web Application**
+
+Optionally, set a name for the application such as ``SkyPilot API Server``. Then, set the following settings:
+
+* Set the **Sign-in redirect URIs** to ``<ENDPOINT>/oauth2/callback``, where ``<ENDPOINT>`` is your API server endpoint.
+  * e.g. ``http://a.b.c.d/oauth2/callback``
+* Set **Assignments > Controlled access** to **Allow everyone in your organization to access**, unless you want to limit access to select groups.
+
+Click **Save**. You will need the Client ID and a Client Secret in the next step.
+
+Deploy in Helm
+~~~~~~~~~~~~~~
+
+Set up the environment variables for your API server deployment. ``NAMESPACE`` and ``RELEASE_NAME`` should be set to the currently installed namespace and release:
+
+.. code-block:: bash
+
+    NAMESPACE=skypilot # TODO: change to your installed namespace
+    RELEASE_NAME=skypilot # TODO: change to your installed release name
+    VERSION=1.0.0-dev20250410 # TODO: change to the version you want to upgrade to
+    IMAGE_REPO=berkeleyskypilot/skypilot-nightly
+
+Use ``helm upgrade`` to redeploy the API server helm chart with the ``skypilot-oauth2-proxy`` deployment. Replace ``<CLIENT ID>`` and ``<CLIENT SECRET>`` with the values from the Okta admin console above.
+
+.. code-block:: console
+
+    $ # --reuse-values is critical to keep the old values that aren't being updated here.
+    $ helm upgrade -n $NAMESPACE $RELEASE_NAME skypilot/skypilot-nightly --devel --reuse-values \
+      --set ingress.oauth2-proxy.enabled=true \
+      --set ingress.oauth2-proxy.client-id=<CLIENT ID> \
+      --set ingress.oauth2-proxy.client-secret=<CLIENT SECRET>
+
+To make sure it's working, visit your endpoint URL in a browser. You should be redirected to Okta to sign in.
+
+Now, you can use ``sky api login -e <ENDPOINT>`` to go though the login flow for the CLI.
+
+.. _auth-proxy-byo:
+
+Bring your own auth proxy
```

Comment:
This is optional, irght? If so, we can put a `Optional: ` in the front of the title.